corehq/apps/app_manager/xpath_validator/wrapper.py

```diff
@@ -16,10 +17,13 @@ def validate_xpath(xpath, allow_case_hashtags=False):
             cmd = ['node', path, '--allow-case-hashtags']
         else:
             cmd = ['node', path]
-        p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE)
-        # Collapse whitespace. '\r' mysteriously causes the process to hang in python 3.
-        stdout, stderr = p.communicate(re.sub(r'\s+', ' ', xpath).encode('utf-8'))
+        if six.PY3:
+            p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE, universal_newlines=True, encoding='utf-8')
+            stdout, stderr = p.communicate(xpath)
+        else:
+            p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            stdout, stderr = p.communicate(re.sub(r'\s+', ' ', xpath).encode('utf-8'))
```

Comment:
if this is for python2, why do we need to do the `re.sub`?

Reply:
good point, updated